web3.js/src/layout.ts

```diff
@@ -15,11 +15,36 @@ export const uint64 = (property: string = 'uint64') => {
   return BufferLayout.blob(8, property);
 };
 
+interface IRustStringShim
+  extends Omit<
+    BufferLayout.Structure<
+      Readonly<{
+        length: number;
+        lengthPadding: number;
+        chars: Uint8Array;
+      }>
+    >,
+    'decode' | 'encode' | 'replicate'
+  > {
+  alloc: (str: string) => number;
+  decode: (b: Uint8Array, offset?: number) => string;
+  encode: (str: string, b: Uint8Array, offset?: number) => number;
+  replicate: (property: string) => this;
+}
+
```

Comment:
There's the data that the _internal_ buffer encodes and decodes, but then there's the encoding/decoding behavior that this exposes externally. This shim describes the public behavior.

Reply:
Nice